.github/workflows/iac-validate.yml

```
@@ -0,0 +1,30 @@
+name: IAC Validation
+on:
+  workflow_call:
+
+jobs:
+  iac-validate:
+    name: IaC Validation
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    defaults:
+      run:
+        working-directory: "./terraform"
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v3
+
+      - name: Setup Terraform
+        uses: hashicorp/setup-terraform@v1
```

Can you test v2 that is working for us?
```suggestion
        uses: hashicorp/setup-terraform@v2
```